gym_graph_traffic/envs/intersection.py

```diff
@@ -108,12 +148,879 @@ def update(self) -> None:
             if self.updates_until_state_change == 0:
                 self.state = "lr"
 
-    def can_i_go(self, from_idx) -> int:
-        (source, dest) = self.dest_dict.get(from_idx, (None, None))
-        if source in self.state and dest is not None:
-            return dest.free_init_cells
+    def can_i_go(self, from_idx, cars_indices):
```

Comment:
I need more explanation offline how this new model works (and then I might be able to review this class, but right now it seems overcomplicated and too reliant on ifs)